examples/error_boundary/public/plugin.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { EuiButton } from '@elastic/eui';
+
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+
+import { AppMountParameters, CoreSetup, CoreStart, Plugin } from '@kbn/core/public';
+import { DeveloperExamplesSetup } from '@kbn/developer-examples-plugin/public';
+import { KibanaErrorBoundary, KibanaErrorBoundaryProvider } from '@kbn/shared-ux-error-boundary';
+import { KibanaPageTemplate } from '@kbn/shared-ux-page-kibana-template';
+
+interface SetupDeps {
+  developerExamples: DeveloperExamplesSetup;
+}
+
+const useErrors = () => {
+  return useState(false);
+};
+
+export const FatalComponent = () => {
```

Comment:
I think using an app in an example plugin, makes sense for functional testing for fatal error handling.

Recoverable error handling, on the other hand should be tested against user-facing pages. That can be achieved using an interceptor that proxies requests, and in testing we intercept chunks of code to return 404, 502, 503, etc. Filed: https://github.com/elastic/kibana/issues/170777